gjstest/internal/js/mock_instance.js

```diff
@@ -148,9 +148,26 @@ gjstest.internal.createMockOfPrototype_ =
   Tmp.prototype = prototype;
   var result = new Tmp;
 
+  // Object.getOwnPropertyNames does not include inherited methods. This method
+  // crawls up the inheritance chain and records all properties.
+  var inheritedPropertyNames = function(obj) {
+    var props = {};
+    while(obj) {
+      // We've reached the base Object - break.
+      if (obj.constructor.name == 'Object') {
+        break;
+      }
+      Object.getOwnPropertyNames(obj).forEach(function(p) {
+        props[p] = true;
+      });
+      obj = Object.getPrototypeOf(obj);
+    }
+    return Object.getOwnPropertyNames(props);
+  };
+
   // Mock each function in the prototype.
-  for (var name in result) {
```

Comment:
So IIUC, this will cause a behavior change, since `Object.getOwnPropertyNames` [includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyNames) non-enumerable properties, right? (Is it possible to write a test for this?)